ql/src/java/org/apache/hadoop/hive/ql/io/parquet/serde/ArrayWritableObjectInspector.java

```diff
@@ -139,8 +160,9 @@ public Object getStructFieldData(final Object data, final StructField fieldRef)
     if (data instanceof ArrayWritable) {
       final ArrayWritable arr = (ArrayWritable) data;
       final StructFieldImpl structField = (StructFieldImpl) fieldRef;
-      if (structField.getIndex() < arr.get().length) {
-        return arr.get()[structField.getIndex()];
+      int index = isRoot ? structField.getIndex() : structField.adjustedIndex;
```

Comment:
All ```StructField```s have two indexes: ```index``` for original index and ```ajustedIndex``` for index after pruned. For those columns doesn't involved in pruned, original index is the same as adjusted index. How about refactoring index-> writeIndex and adjustedIndex -> writeIndex? We can always return ```writeIndex``` getting rid of the ```isRoot```. 
Any thoughts about it?